src/regex.rs

```diff
@@ -14,18 +14,87 @@ impl Regex {
     fn from_pattern(_cls: &PyType, pattern: &str) -> PyResult<Self> {
         libparsec::types::regex::Regex::from_pattern(pattern)
             .map(Regex)
-            .map_err(|_| PyValueError::new_err(format!("Invalid pattern `{}`", pattern)))
+            .map_err(|err| {
+                PyValueError::new_err(format!(
+                    "Failed to convert pattern to regex: `{}` with the following error `{}`",
+                    pattern, err
+                ))
+            })
     }
 
     #[classmethod]
     fn from_regex_str(_cls: &PyType, regex_str: &str) -> PyResult<Self> {
         libparsec::types::regex::Regex::from_regex_str(regex_str)
             .map(Regex)
-            .map_err(|_| PyValueError::new_err(format!("Invalid regex `{}`", regex_str)))
+            .map_err(|err| {
+                PyValueError::new_err(format!(
+                    "Regex creation failed: `{}` with the following error `{}`",
+                    regex_str, err
+                ))
+            })
     }
 
     #[getter]
     fn pattern(&self) -> PyResult<&str> {
         Ok(self.0.as_ref())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use pyo3::{
+        exceptions::PyValueError,
+        types::{PyModule, PyTuple},
+        Py, PyAny, PyResult, Python,
+    };
+
+    const NEW_REGEX_STR: &str = r"^.*\.tmp$|^.*\.temp$|^.*\.swp$|^.*~$|^\.fuse_hidden.*$|^\.directory$|^\.Trash-.*$|^\.nfs.*$|^\.goutputstream-.*$|^\.DS_Store$|^\.AppleDouble$|^\.LSOverride$|^\._.*$|^\.DocumentRevisions-V100$|^\.Spotlight-V100$|^\.TemporaryItems$|^\.Trashes$|^\.VolumeIcon\.icns$|^\.com\.apple\.timemachine\.donotpresent$|^\.AppleDB$|^\.AppleDesktop$|^Network Trash Folder$|^Temporary Items$|^\.apdisk$|^\.fseventsd$|^.*\..*\.sb-.*-.*$|^Thumbs\.db$|^Thumbs\.db:encryptable$|^ehthumbs\.db$|^ehthumbs_vista\.db$|^.*\.stackdump$|^[Dd]esktop\.ini$|^\$RECYCLE\.BIN$|^.*\.lnk$|^\.~.*$|^~\$.*$";
```

Comment:
Maybe split it over several lines?

Reply:
@TimeEngineer How would you do that?

Reply:
with the `concat!` macro ;)

Reply:
Or maybe regenerate it from the code instead of hard coding it?